torch/distributed/rpc/internal.py

```diff
@@ -136,5 +146,30 @@ def _load_return_value(binary_data, tensor_table):
     return result
 
 
+def _start_record_function(exec_type, func_name, current_worker_name, dest_worker_name):
+    """
+    This function should be called from RPC/RRef functions to create a
+    RecordFunction object for profiling. This function also runs the before
+    callbacks that start the profiling, though the user is responsible for
+    running the appropriate callbacks when the function to be profiled finishes.
```

Comment:
What does the last sentence mean? 

> though the user is responsible for running the appropriate callbacks when the function to be profiled finishes.

Does this have any API implication on user side, or is the users here actually referring to our rpc/rref implementations?

Reply:
The actual user should just be able to do:
```
with torch.autograd.profiler.profile() as prof:
# RPC code
```
just like regular autograd profiling. I was referring to the code where we'd have to call `rf->end()` which we call in `Future.h`.